extensions/llama/python/llama.py

```diff
@@ -24,35 +25,57 @@ async def serialize(
         **kwargs,
     ) -> Prompt:
         parameters = parameters or {}
-        out = Prompt(name="TODO", input="TODO")
-        return out
+        out = Prompt(name=prompt_name, input=data)
+        return [out]
 
     async def deserialize(
         self, prompt: Prompt, aiconfig: AIConfigRuntime, params: dict | None = None
     ) -> dict:
-        out = {}
-        return out
+        resolved = resolve_prompt(prompt, params, aiconfig)
+
+        try:
+            remember_chat_context = prompt.metadata.remember_chat_context
+        except AttributeError:
+            remember_chat_context = False
+
+        model_input = (
+            f"CONTEXT:\n{self.history()}\nQUESTION:\n{resolved}"
+            if remember_chat_context
+            else resolved
+        )
+        return {"model_input": model_input}
 
     def id(self) -> str:
         return "LLaMA"
 
+    def history(self) -> str:
+        def _fmt(q, a):
+            return f"Q: {q}\nA: {a}"
+
+        out = "\n".join(_fmt(q, a) for q, a in self.qa)
```

Comment:
Conversation history should be derived from previous prompts of the same model

Reply:
Not sure what you mean exactly

Reply:
The aiconfig runtime will contain a prompt for each message in the history already; we should use that instead of maintaining it in state in the model parser. See the ts implementation for example, or the openai model parser

Reply:
It seems outputs is empty, so I only have the history of questions available. cc @LastMileNewGrad

Reply:
Lol, added a comment on the streaming PR about that -- for now model parsers need to set prompt.outputs within `run_inference` implementation

Reply:
uhh so what's the advantage really? :D It's trivial to just do it like this

Reply:
The advantage is you can take in a config constructed from a conversation, add a new prompt and run that single new prompt -- and it continues the conversation. This current implementation wouldn't do that since it would require manually running all prompts (even though the history is in the config to use)